department-of-veterans-affairs/vets-website
# Direct Deposit EDU migration alerts

## Summary

- Adds alerts around Direct Deposit migration for EDU benefits
- These alerts are hidden by default and are triggered with the corresponding feature toggles:
  - `profile_show_direct_deposit_single_form_alert`
  - `profile_show_direct_deposit_single_form_edu_downtime`
- Refactored some repeated link code into `EligibilityLink.jsx` component
- Changed `FraudVictimAlert` to use the summary box component bc it was not aligned with figma design file
- Updated selectors to align with lighthouse data shape, since there was still some leftover logic from the old endpoint data. Updated selectors unit tests for this change.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/76715

## Testing done

- Created unit tests for this work, along with basic unit/integration tests for the Direct Deposit page altogether since there were none.
- How to test this PR locally
  - When running the local mock server, change either flag to view the corresponding alerts. If the downtime flag is set to true, then it will take precedence over the form_alert flag, so the form alert will be hidden in that case.
  - When a user has both cnp and edu direct deposit and the form_alert flag is true, then they will see the flag.
  - If the user does not have edu, but CNP, then they will not see that alert.
  - If the user is not eligible for either then they should not see the alert
  - To simulate various states of an endpoint you can see example responses that are commented out within the following endpoint sections:
    - `GET /v0/profile/direct_deposits/disability_compensations`
    - `GET /v0/profile/ch33_bank_accounts` 
## Screenshots

There aren't before screenshots because these alerts didn't appear before.

MIGRATION ALERT

|         | Mobile | Desktop |
| ------- | ------ | ----- |
| Hub Alert | ![Screenshot 2024-03-05 at 12 16 45 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/af5e8031-2b58-4018-8ab5-248e0f1f3386) | ![Screenshot 2024-03-05 at 12 15 37 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/ef279c86-2140-4806-965b-347386c179db) |
| Direct Deposit Alert | ![Screenshot 2024-03-05 at 12 17 43 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/d63f3c73-1a37-4a9c-b40e-3d64b2e428e3) | ![Screenshot 2024-03-05 at 12 17 27 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/24c8e493-0e9e-4027-b9e0-b743cd3fb7ed) |

DOWNTIME ALERT

|         | Mobile | Desktop |
| ------- | ------ | ----- |
| Edu of DD page | ![Screenshot 2024-03-05 at 12 24 05 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/1cce054d-9a79-4444-9f70-8dec6bd5a95c) | ![Screenshot 2024-03-05 at 12 23 50 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/c453e362-de97-4171-8320-9dba054e9b13) |

## What areas of the site does it impact?

Profile -> Hub & Direct Deposit pages

## Acceptance criteria

- [x] Alerts are build and can be displayed via feature toggles

### Quality Assurance & Testing


### Error Handling
